snipssonos/snipssonos.py

```diff
@@ -101,141 +100,50 @@ def stop_sonos(self):
+             if (not name):
+                 return
+             name = name[0]
+        for provider in self.providerPlayers:
+            func = getattr(provider, func_name)
+            if func(self.device, name, shuffle):
+                print("playing %s:%s" % (func_name, name))
+                return
+
+    def turn_on_radio(self, radio_name):
+        self.play_template(radio_name, func_name="play_station")
 
     def play_playlist(self, name, _shuffle=False):
-        if self.device is None:
-            return
-        if self.spotify is None:
-            return
-        tracks = self.spotify.get_tracks_from_playlist(name)
-        if tracks is None:
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        if _shuffle:
-            shuffle(tracks)
-        for track in tracks:
-            self.add_from_service(track['track']['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
+        self.play_template(name, _shuffle, "play_playlist")
 
     def play_artist(self, name):
-        if self.device is None:
-            return
-        if self.spotify is None:
-            return
-        tracks = self.spotify.get_top_tracks_from_artist(name)
-        if tracks is None:
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        for track in tracks:
-            self.add_from_service(track['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
+        self.play_template(name, func_name= "play_artist")
 
-    def play_album(self, album, _shuffle=False):
-        if self.device is None:
-            return
-        if self.spotify is None:
-            return
-        tracks = self.spotify.get_tracks_from_album(album)
-        if tracks is None:
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        if _shuffle:
-            shuffle(tracks)
-        for track in tracks:
-            self.add_from_service(track['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
+    def play_album(self, name, _shuffle=False):
+        self.play_template(name, func_name= "play_album")
 
     def play_song(self, name):
-        if self.device is None:
-            return
-        if self.spotify is None:
-            return
-        track = self.spotify.get_track(name)
-        if track is None:
-            return None
-        self.device.stop()
-        self.device.clear_queue()
-        self.add_from_service(track['uri'], self.spotify_service, True)
-        self.device.play_from_queue(0)
+        self.play_template(name, func_name= "play_track")
 
     def play_next_item_in_queue(self):
         if self.device is None:
             return
         try:
             self.device.next()
         except Exception:
-            print "Failed to play next item, maybe last song?"
+            print("Failed to play next item, maybe last song?")
```

Comment:
Exceptions are not for normal behaviour. The last song is a normal behaviour...

Reply:
I can't do anything else